trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/library/StdlibLibraryModel.java

```diff
@@ -0,0 +1,500 @@
+		final Expression maxValue =
+				mTypeSizes.constructLiteralForIntegerType(loc, cType, mTypeSizes.getMaxValueOfPrimitiveType(cType));
+
+		final Expression biggerMinInt = mExpressionTranslation.constructBinaryComparisonExpression(loc,
+				IASTBinaryExpression.op_lessEqual, minValue, cType, expr, cType);
+		final Expression smallerMaxValue = mExpressionTranslation.constructBinaryComparisonExpression(loc,
+				IASTBinaryExpression.op_lessEqual, expr, cType, maxValue, cType);
+		final AssumeStatement inRange = new AssumeStatement(loc, ExpressionFactory.newBinaryExpression(loc,
+				BinaryExpression.Operator.LOGICAND, biggerMinInt, smallerMaxValue));
+		resultBuilder.addStatement(inRange);
+
+		assert CTranslationUtil.isAuxVarMapComplete(mNameHandler, resultBuilder.getDeclarations(),
+				resultBuilder.getAuxVars());
+		return resultBuilder.build();
+	}
+
+	private Result handleAbs(final IDispatcher main, final IASTFunctionCallExpression node, final ILocation loc,
+			final String name, final CPrimitive resultType) {
+		final IASTInitializerClause[] arguments = node.getArguments();
+		mHelper.checkArguments(loc, 1, name, arguments);
+		final ExpressionResultBuilder builder = new ExpressionResultBuilder();
+		final ExpressionResult argResult =
+				mExprResultTransformer.transformDispatchDecaySwitchRexBoolToInt(main, loc, arguments[0]);
+		builder.addAllExceptLrValue(argResult);
+		final Expression expr = argResult.getLrValue().getValue();
+		// abs(MIN_INT) does overflow, so add an assertion for overflow checking
+		if (mOverflowCheckMode != CheckMode.IGNORE && resultType.isIntegerType()
+				&& !mTypeSizes.isUnsigned(resultType)) {
+			final Expression minInt = mTypeSizes.constructLiteralForIntegerType(loc, resultType,
+					mTypeSizes.getMinValueOfPrimitiveType(resultType));
+			final Expression biggerMinInt = mExpressionTranslation.constructBinaryComparisonExpression(loc,
+					IASTBinaryExpression.op_greaterThan, expr, resultType, minInt, resultType);
+			mExpressionTranslation.addOverflowCheck(loc, biggerMinInt, builder);
+		}
+		// Construct if x > 0 then x else -x as LrValue for abs(x)
+		final Expression positive = mExpressionTranslation.constructBinaryComparisonExpression(loc,
+				IASTBinaryExpression.op_greaterThan, expr, resultType,
+				mTypeSizes.constructLiteralForIntegerType(loc, resultType, BigInteger.ZERO), resultType);
+		final Expression negated =
+				mExpressionTranslation.constructUnaryExpression(loc, IASTUnaryExpression.op_minus, expr, resultType);
+		final Expression iteExpression = ExpressionFactory.constructIfThenElseExpression(loc, positive, expr, negated);
+		return builder.setLrValue(new RValue(iteExpression, resultType)).build();
+	}
+
+	private TypesResult handleSize_t(final IASTNamedTypeSpecifier node, final ILocation loc) {
+		return new TypesResult(new PrimitiveType(loc, BoogieType.TYPE_REAL, SFO.REAL), node.isConst(), false,
```

Comment:
I just found that we use Boogie's `real` as a type here, but `size_t` is an `int` type of course. This seems like a bug that has been present quite long in `TypeHandler` (not sure if it actually triggered something)

Reply:
Huh, strange, but I would also think that this is a bug.